src/react-app-env.d.ts

```diff
@@ -1 +1,10 @@
 /// <reference types="react-scripts" />
+
+declare namespace NodeJS {
+  interface ProcessEnv {
+    //types of envs
+    NODE_ENV: "development" | "production" | "test";
+    PUBLIC_URL: string;
+    REACT_APP_API_URL: string;
+  }
+}
```

Comment:
Why do you need this?

Reply:
This is based on [this answer](https://stackoverflow.com/a/53981706/18866062). Without defining the type of `REACT_APP_API_URL`, Typescript gives an error when I use it in `fetch()` because it doesn't know that it's a `string`.

Reply:
Have you tried something like:
```ts
const url: string = process.env.REACT_APP_API_URL;
```
OR
```js
const url = `${process.env.REACT_APP_API_URL}`;
```

Reply:
The first option gives an error which says that `string` or `undefined` cannot be assigned to `string`. The second one works.
To clarify, the file was created when I chose TypeScript as the template for the CRA. So I only had to add the type of `REACT_APP_API_URL`.